src/Psr/CacheItemPool/CacheItem.php

```diff
@@ -164,10 +162,12 @@ public function expiresAfter($time)
      */
     public function getTtl()
     {
-        $ttl = $this->ttl;
-        if ($this->expiration !== null) {
-            $ttl = $this->expiration - time();
+        if ($this->expiration === null) {
+            return null;
         }
-        return $ttl;
+
+        $now = new DateTimeImmutable("now", $this->utc);
```

Comment:
Not important, but we should consider using a clock in future 🤔

Reply:
Was thinking the same when working on it. 
Will create an issue to keep track of it.